subprojects/tooling-api-builders/src/main/java/org/gradle/tooling/internal/provider/runner/BuildModelActionRunner.java

```diff
@@ -47,6 +46,9 @@ public void run(BuildAction action, final BuildController buildController) {
         final GradleInternal gradle = buildController.getGradle();
         gradle.addBuildListener(new BuildResultAdapter(gradle, buildController, buildModelAction));
 
+        if (!buildModelAction.isRunTasksOnly()) {
```

Comment:
❌ The double negation here took me a while to understand.

How about `if (buildModelAction.isModelRequest()) { ... }` instead?